### PR TITLE
Vaibhav feature checkout page

### PR DESCRIPTION
**Pull request by Vaibhav-Kambli**
**Created checkout page for orders**
- Added stepper for checkout component
- Added forms to save shipping address, booking date, time and special instructions from the customer
- Added review page to stepper for reviewing order details
- Added stripe to stepper as the final step of checkout
- Added constants, actions, reducers and context for order booking details

TODO:
- [x] Create Order model for database
- [x] Save order to the database on successful payment
- [x] Update checkout and stripe based on order items from context

TESTING:
- Go to "/payment"
- Input stepper details
- Stripe card :  "4242 4242 4242 4242" MM/YY: "04 / 24" CVV: "242" ZIP: "42424"
- On successful payment, order entry is created inside backend Order DB

Order

### CODE REVIEW

File: client/src/actions/cartActions.js
Comment (WhoaNellie) [inline client/src/actions/cartActions.js]:
```suggestion
```

File: client/src/actions/cartActions.js
Comment (WhoaNellie) [inline client/src/actions/cartActions.js]: Would it be possible to use state/context rather than storing this in LS? Also is the shipping address something that will eventually be stored in the database?

Comment (Vaibhav-Kambli) [reply client/src/actions/cartActions.js]: Yes, I do have it saved inside the state. LS is used to have it persistent with the state. And the shipping address will be saved inside our Order database

Comment (dproc96) [reply client/src/actions/cartActions.js]: I agree with Nellie, I don't think it should be necessary to store this in Local Storage, if you have it in the DB and in state you won't need it in Local Storage

Comment (Vaibhav-Kambli) [reply client/src/actions/cartActions.js]: I will do the necessary changes. The thought process behind putting it in LS was if the user decides to go back and add more items to the cart after putting his booking details then he/she does not have to go through the entire process of adding the details again inside checkout page since state is not persistent. The shipping address is added to the Order DB on successful payment not before that. I will remove LS and update it.

Comment (MinaKhamesi) [reply client/src/actions/cartActions.js]: @Vaibhav-Kambli , I guess we can add address and other cart related info into the cart field that I worked on. We only need a route to change them-it could be a single route both for adding and editing- and that's it I guess? maybe we will need new actions but as far as backend, I already have a route that returns all the info in the cart.

my branch has been approved, after our standup I can merge that and you can add your fields into my cart context? and add actions to save the info to backend. Or I can do that.

 If we could write separate controllers for only those fields that would be ideal, because the edit controller that I have right now is already pretty big function. you can see that in my branch.  Maybe it's best to have routes only for manipulating address and all the fields you were working on.

Comment (Vaibhav-Kambli) [reply client/src/actions/cartActions.js]: @MinaKhamesi I think the changes I have done now should be fine. Because the payment page is already huge since it has stripe functionality and stepper, I cannot really separate the stripe component into it's own file since the checkout button is connected to my stripe form. Also, if I add  functionality to get this data from the backend then it will get even bigger. For now this is working properly. In future we can have it refactored if we move stripe to it's own component

File: server/models/orderModel.js
Comment (MinaKhamesi) [inline server/models/orderModel.js:19]: @Vaibhav-Kambli I really think we need a chefId somewhere, otherwise fetching the orders of a chef won't be possible. I think since each order is assign to a chef, you can add that in your model. In frontend, you can get that information from cart and then pass it on along your body. 
if we decide to rely only on recipeId. we need to fetch all data from this collection populate the recipe field with its info and only from there we could have the info for the chef.which definitely is not a good approach.

Comment (Vaibhav-Kambli) [reply server/models/orderModel.js:19]: Yes, I will add the chef id based on the state value coming from your context.

**Review by MinaKhamesi — APPROVED** [review]
Everything seems to be working perfectly fine. Just wanted to say that I added all constants into userConstant including cart constants. it seems you have another file for that and when you want to work with them you have to bring them from both places. If you had time to make that consistent with my code that would be ideal, otherwise I think everything is working and its best to have it on dev as soon as possible.

**Comment by Vaibhav-Kambli** [comment]
> Everything seems to be working perfectly fine. Just wanted to say that I added all constants into userConstant including cart constants. it seems you have another file for that and when you want to work with them you have to bring them from both places. If you had time to make that consistent with my code that would be ideal, otherwise I think everything is working and its best to have it on dev as soon as possible.

I think it's best to have constants separated because it makes it more modular and easy to track. For now I think this should be fine, we need to get this into dev asap. If I change this, I also need to put orderConstants inside userConstants. Again lot of import changes in different files. If you don't mind we can have this as it is for now, it is properly structured as well.

File: server/models/orderModel.js
Comment (MinaKhamesi) [inline server/models/orderModel.js:25]: @Vaibhav-Kambli you need to have a ref for the chefId like all other object type field

**Review by rudream — APPROVED** [review]
Works great for me!

**Pull request merged**
